src/cascadia/TerminalSettingsModel/Command.cpp

```diff
@@ -149,9 +149,9 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
             return nullptr;
         }
 
-        // Only get the icon path right now. The icon needs to be resolved into
-        // an IconSource on the UI thread, which will be done by RefreshIcon.
-        JsonUtils::GetValueForKey(json, IconKey, result->_IconPath);
+        hstring icon;
+        JsonUtils::GetValueForKey(json, IconKey, icon);
+        result->Icon(icon);

```

Comment:
`OBSERVABLE_GETSET_PROPERTY` makes `_Icon` const.

<details>
<summary>OBSERVABLE_GETSET_PROPERTY Definition</summary>

```c++
#define OBSERVABLE_GETSET_PROPERTY(type, name, event, ...)                             \
public:                                                                                \
    type name() { return _##name; };                                                   \
    void name(const type& value)                                                       \
    {                                                                                  \
        if (_##name != value)                                                          \
        {                                                                              \
            const_cast<type&>(_##name) = value;                                        \
            event(*this, Windows::UI::Xaml::Data::PropertyChangedEventArgs{ L#name }); \
        }                                                                              \
    };                                                                                 \
                                                                                       \
private:                                                                               \
    const type _##name{ __VA_ARGS__ };                                                 \
    void _set##name(const type& value)                                                 \
    {                                                                                  \
        const_cast<type&>(_##name) = value;                                            \
    };
```
</details>

So I had to do this dumb thing. We're going to be hitting this problem _a lot_ with data binding. Does JsonUtils already support overriding setters? Or is there any guidance you can provide here? Like, could we override `=` operator maybe?

Reply:
oh right, it makes it const as a trick to stop you from editing the member without going through the setter.

We can't override the `=` operator because ... we would need a special new struct type for every settable thing.

hmmmmmmm.

well this ain't great.

Reply:
you can drop the const from the member, or you can completely defeat the purpose of JsonUtils and do a...

```
GetValueForKey(o, "K", const_cast<hstring&>(_Icon))
```

which seems worse because of the repeated type ;P

Reply:
Going with the first route then haha. Thanks!